specifyweb/frontend/js_src/lib/components/DataModel/collectionApi.ts

```diff
@@ -154,6 +161,106 @@ export const LazyCollection = Base.extend({
   },
 });
 
+export const IndependentCollection = LazyCollection.extend({
+  __name__: 'IndependentCollectionBase',
+  constructor(options, records = []) {
+    this.table = this.model;
+    assert(_.isArray(records));
+    Base.call(this, records, options);
+    this.filters = options.filters || {};
+    this.domainfilter =
+      Boolean(options.domainfilter) &&
+      this.model?.specifyTable.getScopingRelationship() !== undefined;
+
+    this.removed = new Set<string>();
+    this.updated = {};
+  },
+  initialize(_tables, options) {
+    this.on(
+      'change',
+      function (resource: SpecifyResource<AnySchema>) {
+        if (!resource.isBeingInitialized()) {
+          this.updated[resource.cid] = resource;
+          this.trigger('saverequired');
+        }
+      },
+      this
+    );
+
+    this.on(
+      'add',
+      function (resource: SpecifyResource<AnySchema>) {
+        if (!resource.isNew()) {
+          (this.removed as Set<string>).delete(resource.url());
+          this.updated[resource.cid] = resource.url();
+        } else {
+          this.updated[resource.cid] = resource;
+        }
+        this._totalCount += 1;
+        this.trigger('saverequired');
+      },
+      this
+    );
+
+    this.on(
+      'remove',
+      function (resource: SpecifyResource<AnySchema>) {
+        if (!resource.isNew()) {
+          (this.removed as Set<string>).add(resource.url());
+        }
+        this.updated = removeKey(this.updated, resource.cid);
+        this._totalCount -= 1;
+        this.trigger('saverequired');
+      },
+      this
+    );
+
+    this.listenTo(options.related, 'saved', function () {
+      this.updated = {};
+      this.removed = new Set<string>();
+    });
+
+    setupToOne(this, options);
+  },
+  parse(resp) {
+    const self = this;
+    const records = Reflect.apply(
+      LazyCollection.prototype.parse,
+      this,
+      arguments
+    );
+
+    this._totalCount -= (this.removed as Set<string>).size;
+
+    return records.filter(
+      ({ resource_uri }) => !(this.removed as Set<string>).has(resource_uri)
+    );
+  },
+  async fetch(options) {
+    if (this.related.isBeingInitialized()) {
+      return this;
+    }
+    this.filters[this.field.name.toLowerCase()] = this.related.id;
+
+    const offset =
+      this.length === 0 && this.removed.size > 0
+        ? this.removed.size
+        : this.length;
+
+    options = { ...(options ?? {}), silent: true, offset };
+
+    return Reflect.apply(LazyCollection.prototype.fetch, this, [options]);
```

Comment:
While this does work, a more simple solution be
1. Defining a constant `OFFSET = 20`
2. Have a local variable per collection defining the number of times fetch has been called
3. Incrementing offset based on multiple of that variable

Benefits:
1. This will remove the need for `parse(resp)` --- a record not fetched couldn't have been removed.
2. Don't need to consider `removed.size` in offset calculation.
3. Faster fetches (right now removed resources are still fetched -- quite unepected)

If you don't like variable, you could use a generator (see something similar on backend: 
https://github.com/specify/specify7/blob/aa11b35ac6d282ef97b72e540f9e34d11456fea9/specifyweb/workbench/upload/predicates.py#L256-L262